addons/godot-firebase/database/reference.gd

```diff
@@ -133,13 +136,22 @@ func get_data() -> Dictionary:
     return _store.get_data()
 
 func _get_remaining_path(is_push : bool = true) -> String:
+    var remaining_path = ""
     if !_filter_query or is_push:
-        return _json_list_tag + _query_tag + _auth_tag + Firebase.Auth.auth.idtoken
+        remaining_path = _json_list_tag + _query_tag + _auth_tag + Firebase.Auth.auth.idtoken
     else:
-        return _json_list_tag + _query_tag + _get_filter() + _filter_tag + _auth_tag + Firebase.Auth.auth.idtoken
+        remaining_path = _json_list_tag + _query_tag + _get_filter() + _filter_tag + _auth_tag + Firebase.Auth.auth.idtoken
+    
+    if Firebase.emulating:
+        remaining_path += "&ns="+_config.projectId+"-default-rtdb"
+    
+    return remaining_path
 
-func _get_list_url() -> String:
-    return Firebase.Database._base_url
+func _get_list_url(with_port:bool = true) -> String:
+    var url = Firebase.Database._base_url.trim_suffix(_separator)
+    if with_port and Firebase.emulating:
+        url += ":" + _config.emulators.ports.realtimeDatabase
+    return url + _separator
```

Comment:
defaults to expected behavior, but this could likely be better since with_port is only actually useful when emulating.